HotSpot/Models/TimeInterval.m

```diff
@@ -0,0 +1,50 @@
+//
+//  TimeInterval.m
+//  HotSpot
+//
+//  Created by drealin on 7/19/19.
+//  Copyright © 2019 aodemuyi. All rights reserved.
+//
+
+#import "TimeInterval.h"
+
+@interface TimeInterval()<PFSubclassing>
+@end
+
+@implementation TimeInterval
+{
+    NSDateInterval *dateInterval;
+}
+@dynamic startTime;
+@dynamic endTime;
+@dynamic repeatsWeekly;
+
+# pragma mark - Class Methods
++ (nonnull NSString *)parseClassName {
+    return @"TimeInterval";
+}
+
+# pragma mark - Public Methods
+- (NSDateInterval * _Nullable)intersectionWithTimeInterval:(TimeInterval *)timeInterval {
+    if(!dateInterval) {
+        dateInterval = [[NSDateInterval alloc] initWithStartDate:self.startTime endDate:self.endTime];
+    }
+    NSDateInterval *theirDateInterval = [[NSDateInterval alloc] initWithStartDate:timeInterval.startTime endDate:timeInterval.endTime];
+    if (self.repeatsWeekly) {
```

Comment:
if repeatsWeekly is an NSNumber object, this if statement will return true regardless of whether it stores true or false, because the NSNumber object is not nil. Essentially, you are currently checking if (repeatsWeekly != nil)